File: MemberManagerFrontend/src/model/Member.ts
interface Member {
  id: number,
  name: string,
  server: string,
  avatarUri: string,
  jobInfos: JobInfo[],
  joinedDate: Date,
  promotionDate: Date
}

interface JobInfo {
  name: string,
  level: number,
  classType: string
}

export default Member

File: MemberManagerBackend/src/main/kotlin/com/example/membermanagerbackend/members/model/Member.kt
package com.example.membermanagerbackend.members.model

import org.springframework.data.annotation.Id
import org.springframework.data.mongodb.core.mapping.Document
import java.time.LocalDate

@Document
data class Member(
    @Id
    var characterId: Int = 0,
    val name: String,
    val server: String,
    var avatarUri: String = "",
    var jobInfos: MutableSet<MemberJobInfo> = mutableSetOf(),
    var joinedDate: LocalDate,
    var promotionDate: LocalDate? = null
)
